java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -315,6 +321,32 @@ public LanceScanner newScan(ScanOptions options) {
     }
   }
 
+  /**
+   * Select rows of data by index.
+   *
+   * @param indices the indices to take
+   * @param columns the columns to take
+   * @return an ArrowReader
+   */
+  public ArrowReader take(List<Integer> indices, List<String> columns) throws IOException {
+    try (LockManager.ReadLock readLock = lockManager.acquireReadLock()) {
+      Preconditions.checkArgument(nativeDatasetHandle != 0, "Scanner is closed");
+      byte[] arrowData = nativeTake(indices, columns);
```

Comment:
Can use the stream read to avoid huge memory allocate?

Reply:
Sounds reasonable, let me have a try.

Reply:
When I have a glance, there is an `experimental` API named `take_scan` that may be designed for a big number take/scan. It receives a streaming `row_ranges` and returns a `DatasetRecordBatchStream`. 

It would be better to provide another take method in the Java module for big-take-oriented optimized API.